Controllers/PostsController.cs

```diff
@@ -9,113 +10,63 @@
 
 namespace blog_net_core.Controllers
 {
-    [Route("api/[controller]")]
+    [Route("[controller]")]
     [ApiController]
     public class PostsController : ControllerBase
     {
-        private BLOGContext _context;
-        public PostsController(BLOGContext context)
+        private readonly IBlogService _blogService;
+
+        public PostsController(IBlogService blogService)
         {
-            _context = context;
+            _blogService = blogService;
```

Comment:
its the postService, the blog could be an entity that manage the relations of many posts